pkg/deploy/clusterconfig.go

```diff
@@ -17,6 +17,24 @@ import (
 	corev1 "k8s.io/api/core/v1"
 )
 
+func arePodsOfIPFamily(agentPods []*corev1.Pod, ipFamily string) bool {
+	for _, p := range agentPods {
+		if len(p.Status.PodIPs) == 1 {
+			podIP := net.ParseIP(p.Status.PodIPs[0].IP)
+			if podIP == nil {
+				return false
+			}
+			if ipFamily == "IPv4" && podIP.To4() == nil {
+				return false
+			}
+			if ipFamily == "IPv6" && podIP.To4() != nil {
+				return false
+			}
+		}
+	}
+	return true
+}
```

Comment:
I'm currently playing around with this PR and was a bit confused about this function.

Does it assume that if a pod has more than one IP that it must be a dual-stack pod? What about, arguably a very strange edge case, pods with multiple IPv4 addresses?

Reply:
Also, is it intended that `podIP.To4()` is called for both cases ("IPv4" and "IPv6")?

Reply:
Good point! I adapted the function to check all IPs.